boards/sensoan/sgw2/Kconfig

```diff
@@ -0,0 +1,56 @@
+# SGW-2 board configuration
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+# Copyright (c) 2024 Sensoan Oy
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_SGW2_NRF9160 || BOARD_SGW2_NRF9160_NS
+
+config BOARD_INIT_PRIORITY
+	int "Initialization priority for board configuration"
+	default 80
+
+endif # BOARD_SGW2_NRF9160 || BOARD_SGW2_NRF9160_NS
+
+if BOARD_SGW2_NRF5340_CPUAPP || BOARD_SGW2_NRF5340_CPUAPP_NS || BOARD_SGW2_NRF5340_CPUNET
+
+if BOARD_SGW2_NRF5340_CPUAPP || BOARD_SGW2_NRF5340_CPUAPP_NS
+
+config DOMAIN_CPUNET_BOARD
+	string
+	default "sgw2/nrf5340/cpunet"
+	help
+	  The board which will be used for CPUNET domain when creating a multi
+	  image application where one or more images should be located on
+	  another board. For example hci_rpmsg on the nRF5340_cpunet for
+	  Bluetooth applications.
+
+endif #  BOARD_SGW2_NRF5340_CPUAPP || BOARD_SGW2_NRF5340_CPUAPP_NS
+
+config DOMAIN_CPUAPP_BOARD
+	string
+	default "sgw2/nrf5340/cpuapp"
+	depends on BOARD_SGW2_NRF5340_CPUNET
+	help
+	  The board which will be used for CPUAPP domain when creating a multi
+	  image application where one or more images should be located on
+	  another board.
+
+endif # BOARD_SGW2_NRF5340_CPUAPP || BOARD_SGW2_NRF5340_CPUAPP_NS || BOARD_SGW2_NRF5340_CPUNET
+
+if DT_HAS_NORDIC_NRF_SPIM_ENABLED && DT_HAS_JEDEC_SPI_NOR_ENABLED
+
+config SGW2_SPI_FLASH
+	bool "Imply config options for external flash"
+	imply FLASH
+	imply MULTITHREADING
+	imply SPI_NOR
+	default y
```

Comment:
don't really see why just having the SPI NOR device in dts should enable SPI NOR, multithreading and flash, nor why there should be a Kconfig for this. If this is for MCUboot then submit a config file for MCUboot directly to the repo